packages/core/src/odata/common/selectable/collection-field.ts

```diff
@@ -18,7 +18,8 @@ import { ConstructorOrField } from './constructor-or-field';
  */
 export class CollectionField<
   EntityT extends EntityBase,
-  FieldT extends EdmTypeShared<'any'> | Record<string, any> = any
+  // todo there is not super class for enum like TestEnumType
```

Comment:
I do not really get this comment. Is it about that there is not proper type we can put for the enums, because each one is dynamically generated?